### ISSUE TITLE
[Bug] draw.io diagrams saved in light mode even when app is in dark mode

### ISSUE THREAD

**Issue by neov5** [docmost/docmost#476]
draw.io diagrams are saved by default in light mode. This leads to them showing black text on a black background, like the following:

<img width="445" alt="Screenshot 2024-11-08 at 11 59 51 PM" src="https://github.com/user-attachments/assets/fe6c098e-b3fc-4ff9-94c8-9d53fce8924e">

some poking around the codebase led me to [this block of code](https://github.com/docmost/docmost/blob/main/apps/client/src/features/editor/components/drawio/drawio-view.tsx):

```
onSave={(data: EventSave) => {
  // If the save is triggered by another event, then do nothing
  if (data.parentEvent !== 'save') {
    return;
  }
  handleSave(data);
}}
```

onSave is exposed by `react-drawio`, which [does this on save](https://github.com/marcveens/react-drawio/blob/main/src/DrawIoEmbed.tsx):

```
save: (data) => {
  action.exportDiagram({
    format: exportFormat || 'xmlsvg',
    // @ts-ignore not allowed normally, but only for internal use
    exit: data.exit,
    parentEvent: 'save'
  });
},
```

A possible fix is to bundle a version of react-drawio with the extra parameter `keepTheme: true` in this save callback, to retain the theme when saving. Other solutions might include some callback magic but my TS is not strong enough for that.

**Comment by neov5** [comment]
react-drawio also has [this issue](https://github.com/marcveens/react-drawio/issues/40) opened recently about exposing these options as props, so maybe if a fix for that gets merged using it would be easier

**Comment by Philipinho** [comment]
If we set `keepTheme` to true, won't light mode users complain of the same thing?
In the case of Excalidraw, I have resorted to saving the diagrams in light mode.

Since the finalized diagram is saved to the file storage, I think it is impossible to satisfy both theme options.

**Comment by neov5** [comment]
You're right that it's impossible to satisfy both theme options, but you can have light mode users' diagrams saved in light mode and dark mode users' saved in dark mode because we set the draw.io theme depending on the current docmost theme.
